src/test/fuzz/util.h

```diff
@@ -338,9 +339,17 @@ inline void FillNode(FuzzedDataProvider& fuzzed_data_provider, CNode& node, cons
     }
 }
 
-inline void InitializeFuzzingContext(const std::string& chain_name = CBaseChainParams::REGTEST)
+template <class T = const BasicTestingSetup>
+inline std::unique_ptr<T> MakeFuzzingContext(const std::string& chain_name = CBaseChainParams::REGTEST, const std::vector<const char*>& extra_args = {})
```

Comment:
style nit: templates are already inline, no need to specify twice

Reply:
Also, it looks like returning `T` is just fine here and should be sufficient to show that the caller owns the object that is returned:

```diff
diff --git a/src/test/fuzz/process_message.cpp b/src/test/fuzz/process_message.cpp
index e9c5cb5954..19523d6951 100644
--- a/src/test/fuzz/process_message.cpp
+++ b/src/test/fuzz/process_message.cpp
@@ -39,7 +39,7 @@ const TestingSetup* g_setup;
 void initialize_process_message()
 {
     static const auto testing_setup = MakeFuzzingContext<const TestingSetup>();
-    g_setup = testing_setup.get();
+    g_setup = &testing_setup;
     for (int i = 0; i < 2 * COINBASE_MATURITY; i++) {
         MineBlock(g_setup->m_node, CScript() << OP_TRUE);
     }
diff --git a/src/test/fuzz/process_messages.cpp b/src/test/fuzz/process_messages.cpp
index 291d5df191..d8fa1028e5 100644
--- a/src/test/fuzz/process_messages.cpp
+++ b/src/test/fuzz/process_messages.cpp
@@ -24,7 +24,7 @@ const TestingSetup* g_setup;
 void initialize_process_messages()
 {
     static const auto testing_setup = MakeFuzzingContext<const TestingSetup>();
-    g_setup = testing_setup.get();
+    g_setup = &testing_setup;
     for (int i = 0; i < 2 * COINBASE_MATURITY; i++) {
         MineBlock(g_setup->m_node, CScript() << OP_TRUE);
     }
diff --git a/src/test/fuzz/util.h b/src/test/fuzz/util.h
index 9f1dfa44a1..68c6838a36 100644
--- a/src/test/fuzz/util.h
+++ b/src/test/fuzz/util.h
@@ -339,8 +339,8 @@ inline void FillNode(FuzzedDataProvider& fuzzed_data_provider, CNode& node, cons
     }
 }
 
-template <class T = const BasicTestingSetup>
-inline std::unique_ptr<T> MakeFuzzingContext(const std::string& chain_name = CBaseChainParams::REGTEST, const std::vector<const char*>& extra_args = {})
+template <class T = BasicTestingSetup>
+T MakeFuzzingContext(const std::string& chain_name = CBaseChainParams::REGTEST, const std::vector<const char*>& extra_args = {})
 {
     // Prepend default arguments for fuzzing
     const std::vector<const char*> arguments = Cat(
@@ -349,7 +349,7 @@ inline std::unique_ptr<T> MakeFuzzingContext(const std::string& chain_name = CBa
         },
         extra_args);
 
-    return MakeUnique<T>(chain_name, arguments);
+    return T{chain_name, arguments};
 }
 
 class FuzzedFileProvider

Reply:
w/re returning `T`, I think I'm going to keep the `unique_ptr` just because in the future we might do something in `MakeFuzzingContext` that doesn't trigger RVO and I'm not sure about the copy semantics of our `*TestingSetup` classes...

Reply:
As you are using `auto`, changing from/to unique_ptr is a small diff, so can be done later easily, if needed